Packs/knowbe4Phisher/Integrations/knowbe4Phisher/knowbe4Phisher.py

```diff
@@ -0,0 +1,812 @@
+    except Exception as e:
+        if 'Unauthorized' in str(e):
+            return 'Authorization Error: make sure API Key was set correctly'
+        elif 'NoneType' in str(e):
+            return 'Check the format of First Fetch Time'
+        elif 'Failed to parse json' in str(e):
+            return 'Please check the specified URL'
+        else:
+            return 'Unknown error, please check your configuration'
+
+
+def phisher_message_list_command(client: Client, args: dict) -> CommandResults:
+    """
+    this function implements the message-list-command.
+    arguments of command:
+    limit - maximum number of messages to retrieve. default - 200
+    query - a lucene query for entered by a user to filter results. default - empty
+    message_id - id of specific message to retrieve, once it's entered the query parameter is ignored.
+    all_events - if set to false message will be retrieved without associated events. on true all events will be
+    retrieved to the XSOAR context. default is false.
+
+    arguments of function:
+    client and arguments
+
+    return: all relevant messages returned in the commandResults structure to war room and context of XSOAR
+
+    """
+    # get parameters
+    limit = args.get('limit')
+    query = args.get('query')
+    message_id = args.get('id')
+    all_events = args.get('include_events')
+    # handle query
+    if not query:
+        query = "\\\"\\\""
+    else:
+        query = "\\\"" + str(args.get('query')) + "\\\""  # type: ignore
+    # handle in case ID is given
+    if message_id:
+        query = "\\\"id:" + message_id + "\\\""
+    # create request body
+    query = query.lower()
+    payload_init = ALL_EVENTS_PAYLOAD.format(limit)
+    payload = create_gql_query(payload_init, query)
+    req = create_request(payload)
+    # call the API
+    result = client.phisher_message_list_all_events_request(req)
+    messages = result.get('data', {}).get('phisherMessages', {}).get('nodes', [])
+    readable = []
+    # creata data for XSOAR
+    for item in messages:
+        # find creation time and create a key with this value
+        events = item.get('events', {})
+        for event in events:
+            if (event['eventType']) == 'CREATED':
+                creation_time = (event['createdAt'])
+                break
+        incident_created_time = dateparser.parse(creation_time)
+        if all_events == "False":
+            item.pop("events")
+        item["created at"] = incident_created_time.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        # prepare printout to war room
+        readable.append({"ID": item.get('id', ""), "Status": item.get('actionStatus', ""), "Category": item.get(
+            'category', ""), "From": item.get('from', ""), "Severity": item.get('severity', "")})
+
+    markdown = '### The requested information:\n'
+    markdown += tableToMarkdown('Messages', readable, headers=['ID', 'Status', 'Category', 'From', 'Severity'])
```

Comment:
I think the title is redundant.
Also you could save this dict construction by doing the following:
```suggestion

    markdown += tableToMarkdown('Messages', camelize(messages), headerTransform=pascalToSpace, headers=['ID', 'Status', 'Category', 'From', 'Severity'])
```

Reply:
I agree that the title is not needed, i removed it
the new table is missing values not sure why:
![image](https://user-images.githubusercontent.com/85438368/129478131-97fadba2-2bbd-4678-a026-734271ec60db.png)

Reply:
agreed to leave as it was initially done